math/line2d.h

```diff
@@ -80,8 +80,16 @@ struct Line {
     } else if (p2d >= l && p3d >= l) {
       return std::min<T>((p2 - p1).norm(), (p3 - p1).norm());
     }
-    const Vector2T n = UnitNormal();
-    return std::min<T>(std::abs<T>(n.dot(p20)), std::abs<T>(n.dot(p30)));
+
+    const Vector2T p02 = p0 - p2;
+    const Vector2T p12 = p1 - p2;
+    const Vector2T this_n = UnitNormal();
+    const Vector2T other_n = Line(p2, p3).UnitNormal();
```

Comment:
Why this roundabout approach? Please change to: `Perp((p3 - p2).normalized())`

Reply:
I used that because it was more consistent with the rest of the code. I can re-do this part, but there's other parts of this function that, at least naively, compute several norms when they could do the same computation with only. I've not checked the ASM to verify this however.